mirrord/agent/src/util.rs

```diff
@@ -86,22 +86,26 @@ where
     }
 }
 
+/// `IndexAllocator` is a helper struct that allocates unique indices for a given type.
+/// It keeps tracked of the indices that were freed, and reuses them when possible.
+/// You can provide a buffer of indices using the second generic `N` to avoid re-using indices
```

Comment:
The solution is a bit weird, and if feels like it's kinda just pushing the potential problem forward. As I think most users don't reach the `100`  indexes that are being used.

What do we need to fully solve this? Have the allocator be aware of multiple threads? Hold the allocator with a `Mutex`? Or maybe we drop the idea of freeing and just keep increasing the index?

I'm ok with approving this if we add a `TODO` with possible better solutions.

Reply:
The problem isn't multiple threads but case when layer doesn't know of re-use by agent then weird stuff can happen.

Reply:
Oh, ok.